File: backend/app/schemas/prompt.py
"""提示词相关的 Pydantic Schemas"""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field
import uuid


class PromptCategory:
    """提示词类别常量"""
    ANALYSIS = "analysis"
    GENERATION = "generation"
    CHECK = "check"


class PromptResponse(BaseModel):
    """提示词响应模型"""
    scene_key: str = Field(..., description="场景标识")
    scene_name: str = Field(..., description="场景名称")
    category: str = Field(..., description="类别：analysis/generation/check")
    prompt: str = Field(..., description="完整提示词")
    available_vars: Optional[dict] = Field(None, description="可用变量列表")
    version: int = Field(..., description="当前版本号")
    is_customized: bool = Field(False, description="是否已自定义（区别于内置默认）")
    updated_at: Optional[datetime] = Field(None, description="最后更新时间")

    model_config = {"from_attributes": True}


class PromptUpdate(BaseModel):
    """提示词更新模型"""
    prompt: str = Field(..., min_length=10, description="完整提示词")


class PromptVersionResponse(BaseModel):
    """提示词版本响应模型"""
    id: uuid.UUID = Field(..., description="版本ID")
    version: int = Field(..., description="版本号")
    prompt: str = Field(..., description="提示词")
    created_by: Optional[uuid.UUID] = Field(None, description="创建者ID")
    created_by_name: Optional[str] = Field(None, description="创建者名称")
    created_at: datetime = Field(..., description="创建时间")

    model_config = {"from_attributes": True}


class PromptVersionListResponse(BaseModel):
    """提示词版本列表响应"""
    items: list[PromptVersionResponse]
    total: int


class PromptListResponse(BaseModel):
    """提示词列表响应"""
    items: list[PromptResponse]
    total: int


class ProjectPromptConfig(BaseModel):
    """项目提示词配置"""
    scene_key: str = Field(..., description="场景标识")
    scene_name: str = Field(..., description="场景名称")
    category: str = Field(..., description="类别")
    # 最终使用的提示词（已解析三层回退）
    prompt: str = Field(..., description="最终使用的提示词")
    available_vars: Optional[dict] = Field(None, description="可用变量列表")
    # 继承状态
    source: str = Field(..., description="提示词来源：builtin/global/project")
    has_project_override: bool = Field(False, description="项目是否有自定义覆盖")
    has_global_override: bool = Field(False, description="全局是否有自定义覆盖")


class ProjectPromptConfigListResponse(BaseModel):
    """项目提示词配置列表响应"""
    items: list[ProjectPromptConfig]
    total: int


class ProjectPromptOverride(BaseModel):
    """项目提示词覆盖设置"""
    prompt: str = Field(..., min_length=10, description="提示词")


class PromptRollbackRequest(BaseModel):
    """提示词回滚请求"""
    version: int = Field(..., ge=1, description="目标版本号")

File: frontend/src/types/prompt.ts
/**
 * 提示词配置相关类型定义
 */

// 提示词类别
export type PromptCategory = 'analysis' | 'generation' | 'check';

// 提示词类别中文名称
export const PROMPT_CATEGORY_NAMES: Record<PromptCategory, string> = {
  analysis: '解析类',
  generation: '生成类',
  check: '检查类',
};

// 提示词响应
export interface PromptResponse {
  scene_key: string;
  scene_name: string;
  category: PromptCategory;
  prompt: string;
  available_vars: Record<string, string> | null;
  version: number;
  is_customized: boolean;
  updated_at: string | null;
}

// 提示词列表响应
export interface PromptListResponse {
  items: PromptResponse[];
  total: number;
}

// 提示词更新请求
export interface PromptUpdate {
  prompt: string;
}

// 提示词版本响应
export interface PromptVersionResponse {
  id: string;
  version: number;
  prompt: string;
  created_by: string | null;
  created_by_name: string | null;
  created_at: string;
}

// 提示词版本列表响应
export interface PromptVersionListResponse {
  items: PromptVersionResponse[];
  total: number;
}

// 项目提示词配置
export interface ProjectPromptConfig {
  scene_key: string;
  scene_name: string;
  category: PromptCategory;
  prompt: string;
  available_vars: Record<string, string> | null;
  source: 'project' | 'global' | 'builtin';
  has_project_override: boolean;
  has_global_override: boolean;
}

// 项目提示词配置列表响应
export interface ProjectPromptConfigListResponse {
  items: ProjectPromptConfig[];
  total: number;
}

// 项目提示词覆盖请求
export interface ProjectPromptOverride {
  prompt: string;
}

// 提示词回滚请求
export interface PromptRollbackRequest {
  version: number;
}

// 场景定义（用于前端显示）
export interface PromptScene {
  key: string;
  name: string;
  category: PromptCategory;
  description: string;
}
